metacat-main/src/main/java/com/netflix/metacat/main/api/v1/MetadataController.java

```diff
@@ -258,7 +238,24 @@ public void deleteDefinitionMetadata(
      */
     @RequestMapping(method = RequestMethod.DELETE, path = "/data/process")
```

Comment:
Should we name it as path = "/datametadata/process"? I am not sure about the 'process', but we should explicitly point out the datametadata and definitionmeatdata part.

Reply:
This is a metadata controller. Hence I named it data or definition.

Reply:
The ending point ( path ) should be self-explained, the client shouldn't know what controller it connects to.

Reply:
How about rename to path = "/obsoleteData" and "/obsoleteDefinition"? I do want to make the ending point self-explained what will actually happen.  The method name 'Delete' is explained the action, but the path /data/process is not clear.

Reply:
I am not good with names but I will rename it to /data/cleanup and /definition/cleanup. Is it fine?

Reply:
I thought about the 'cleanup' too, but it reads like 'delete /data/cleanup', and recommended 'delete obsoletedata'.  Both of them are better than 'process'..

Reply:
What @zhljen is saying here is really more of a RPC vs. REST argument. I see what she's saying about making it clear but I think it breaks consistency with a lot of our API.

If it isn't clear via the RESTful path of a `DELETE` to `/data/process` then that is the fault of the naming of those payloads and we probably should try to fix that root cause rather than hack some RPC method paths into it to get around it.

just my 2 cents.

Reply:
This is not the DELETE for the resource else it would have been /definition or /data with DELETE method. This is not a resource API. It is an API for cleaning up the invalid entries in definition_metadata and data_metadata and hence I named it /process but we could be more precise by saying cleanup..

Reply:
Lets talk about it on Monday in the meeting to come to a good answer for this going forward. @zhljen and I were talking about it in the office and we probably just need to come up with a consistent answer going forward. This seems to be a administration type API and perhaps we need to break these kinds of things apart